dev/create-release/release-build.sh

```diff
@@ -164,7 +164,6 @@ DEST_DIR_NAME="$SPARK_PACKAGE_VERSION"
 
 git clean -d -f -x
 rm .gitignore
-rm -rf .git
```

Comment:
@kiszk . Could you confirm that the release tarballs do not have `.git` inside?

Reply:
why change this?

Reply:
@dongjoon-hyun Here is the result. I confirmed that release tarballs do not have `.git` directory.
```
$ tar xf spark-2.3.4-bin-hadoop2.6.tgz
$ cd spark-2.3.4-bin-hadoop2.6/
$ bin/spark-submit --version
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.3.4
      /_/
                        
Using Scala version 2.11.8, OpenJDK 64-Bit Server VM, 1.8.0_212
Branch HEAD
Compiled by user ishizaki on 2019-09-02T19:30:42Z
Revision 8c6f8150f3c6298ff4e1c7e06028f12d7eaf0210
Url https://gitbox.apache.org/repos/asf/spark.git
Type --help for more information.
$ ls -al
total 120
drwxr-xr-x 13 ishizaki ishizaki  4096 Sep  3 04:43 .
drwxrwxr-x  5 ishizaki ishizaki  4096 Sep  3 10:31 ..
-rw-r--r--  1 ishizaki ishizaki 18045 Sep  3 04:43 LICENSE
-rw-r--r--  1 ishizaki ishizaki 26366 Sep  3 04:43 NOTICE
drwxr-xr-x  3 ishizaki ishizaki  4096 Sep  3 04:43 R
-rw-r--r--  1 ishizaki ishizaki  3809 Sep  3 04:43 README.md
-rw-r--r--  1 ishizaki ishizaki   203 Sep  3 04:43 RELEASE
drwxr-xr-x  2 ishizaki ishizaki  4096 Sep  3 04:43 bin
drwxr-xr-x  2 ishizaki ishizaki  4096 Sep  3 04:43 conf
drwxr-xr-x  5 ishizaki ishizaki  4096 Sep  3 04:43 data
drwxr-xr-x  4 ishizaki ishizaki  4096 Sep  3 04:43 examples
drwxr-xr-x  2 ishizaki ishizaki 16384 Sep  3 04:43 jars
drwxr-xr-x  3 ishizaki ishizaki  4096 Sep  3 04:43 kubernetes
drwxr-xr-x  2 ishizaki ishizaki  4096 Sep  3 04:43 licenses
drwxr-xr-x  6 ishizaki ishizaki  4096 Sep  3 04:43 python
drwxr-xr-x  2 ishizaki ishizaki  4096 Sep  3 04:43 sbin
drwxr-xr-x  2 ishizaki ishizaki  4096 Sep  3 04:43 yarn
$ cd ..
$ tar tvf spark-2.3.4-bin-hadoop2.6.tgz | grep "spark-2.3.4-bin-hadoop2.6/.git"
$
```

Reply:
@felixcheung Without this PR, then tarballs is created by `dev/create-release/do-release-docker.sh`, the version information related to output by `git` command is missing.   
This is because `git` command executed without `.git directory`. As a result, `git` command (e.g. `git git rev-parse HEAD`) returns empty. Then, version information is missing.

This change tries to keep `.git` directory to correctly execute `.git` command in `build/spark-build-info`.

Reply:
> $ tar xf spark-2.3.4-bin-hadoop2.6.tgz

You need to check the source archive, not the binary one.

Reply:
@vanzin you are right. Unfortunately, I found `.git` directory ...   
Let me work more.

```
$ tar xf spark-2.3.4.tgz 
$ ls -al spark-2.3.4
total 312
drwxr-xr-x 31 ishizaki ishizaki   4096 Sep  3 04:07 .
drwxrwxr-x  6 ishizaki ishizaki   4096 Sep  4 03:31 ..
drwxr-xr-x  8 ishizaki ishizaki   4096 Sep  3 04:07 .git
-rw-r--r--  1 ishizaki ishizaki     40 Sep  3 04:07 .gitattributes
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 .github
-rw-r--r--  1 ishizaki ishizaki   1751 Sep  3 04:07 .travis.yml
-rw-r--r--  1 ishizaki ishizaki    995 Sep  3 04:07 CONTRIBUTING.md
-rw-r--r--  1 ishizaki ishizaki  18045 Sep  3 04:07 LICENSE
-rw-r--r--  1 ishizaki ishizaki  26366 Sep  3 04:07 NOTICE
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 R
-rw-r--r--  1 ishizaki ishizaki   3809 Sep  3 04:07 README.md
-rw-r--r--  1 ishizaki ishizaki   2318 Sep  3 04:07 appveyor.yml
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 assembly
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 bin
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 build
drwxr-xr-x  9 ishizaki ishizaki   4096 Sep  3 04:07 common
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 conf
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 core
drwxr-xr-x  5 ishizaki ishizaki   4096 Sep  3 04:07 data
drwxr-xr-x  6 ishizaki ishizaki   4096 Sep  3 04:07 dev
drwxr-xr-x  9 ishizaki ishizaki   4096 Sep  3 04:07 docs
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 examples
drwxr-xr-x 15 ishizaki ishizaki   4096 Sep  3 04:07 external
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 graphx
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 hadoop-cloud
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 launcher
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 licenses
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 licenses-binary
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 mllib
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 mllib-local
-rw-r--r--  1 ishizaki ishizaki 101670 Sep  3 04:07 pom.xml
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 project
drwxr-xr-x  6 ishizaki ishizaki   4096 Sep  3 04:07 python
drwxr-xr-x  5 ishizaki ishizaki   4096 Sep  3 04:07 repl
drwxr-xr-x  5 ishizaki ishizaki   4096 Sep  3 04:07 resource-managers
drwxr-xr-x  2 ishizaki ishizaki   4096 Sep  3 04:07 sbin
-rw-r--r--  1 ishizaki ishizaki  17624 Sep  3 04:07 scalastyle-config.xml
drwxr-xr-x  6 ishizaki ishizaki   4096 Sep  3 04:07 sql
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 streaming
drwxr-xr-x  3 ishizaki ishizaki   4096 Sep  3 04:07 tools
```